lxd/device/nic_bridged.go

```diff
@@ -965,6 +965,13 @@ func (d *nicBridged) removeFilters(m deviceConfig.Device) {
 		logger.Errorf("Failed to remove static IP network filters for %q: %v", d.name, err)
 	}
 
+	// Clear any "block all" filters by passing in empty slices
+	logger.Debug("Clearing filters which drop entire protocols", log.Ctx{"project": d.inst.Project(), "instance": d.inst.Name(), "parent": m["parent"], "dev": d.name, "host_name": m["host_name"], "hwaddr": m["hwaddr"]})
+	err = d.state.Firewall.InstanceClearBridgeFilter(d.inst.Project(), d.inst.Name(), d.name, m["parent"], m["host_name"], m["hwaddr"], make([]*net.IPNet, 0, 4), make([]*net.IPNet, 0, 4))
```

Comment:
When the device has a parent network that is managed but does not have DCHP enabled, the `ipv{n}.address` of the cloned config is set to `none`. This blocks all protocols (as expected) but because the config is not saved to the device, `allowedIPNets` will return `nil` for the `IPv{n}Nets` instead of an empty slice.

Previously, if `ipv{n}.address` was empty or `none`, then `IPv{n}` was set to `firewallDrivers.FilterIPv{n}All` which would have caught this case. 

Looking back at this now, we could do something analogous to the above by checking if the `IPv{n}Nets` are nil, and if they are, set them equal to empty slices. This would remove the extra call.

Reply:
Removing the extra call would be good as its expensive.

Reply:
I've removed the extra call here.